### ISSUE TITLE
System Generated Comments - Extension

### ISSUE THREAD

**Issue by JHarrietha-AOT** [bcgov/foi-flow#876]
* As a (type of user)
* I want (goal) 
* so that (reason/ why/ value) 

**Assumptions & Scope**	
* All user groups/types will be able to view all system generated comments
* System generated comments cannot be edited or deleted
* Username for system generated comments will display as 'Request History'
* Comments will generate for all user types when completing the specific action (i.e. so long as the user has permission to complete the action a comment will be generated)
* For extensions, a comment will only be posted once the extension is approved, denied or deleted. Pending extensions will not generate a comment
* Extension comments will include the number of days extended and the new legislated due date

What is IN scope?

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: Public Body Extension system generated comment**
- GIVEN I am an IAO user (only IAO users have permission to take extensions)
- WHEN I complete a Public Body Extension for a request
- THEN a comment will appear within the Comments tab of the request
- AND the comment will say '[Username] has taken a [XX] day Public Body extension. The new legislated due date is [YYY MMM DD]
- AND the comment will display the timestamp of when the state change was completed
- AND the ellipsis on the right of the comment will be inactive (greyed out)

**Scenario 2: Approved OIPC Extension system generated comment**
- GIVEN I am an IAO user (only IAO users have permission to take extensions)
- WHEN I approve and save an OIPC Extension for a request
- THEN a comment will appear within the Comments tab of the request
- AND the comment will say 'The OIPC has approved a [XX] day extension. The new legislated due date is [YYY MMM DD]
- AND the comment will display the timestamp of when the state change was completed
- AND the ellipsis on the right of the comment will be inactive (greyed out)

**Scenario 3: Denied OIPC Extension system generated comment**
- GIVEN I am an IAO user
- WHEN I deny and save an OIPC Extension for a request
- THEN a comment will appear within the Comments tab of the request
- AND the comment will say 'The OIPC has denied a [XX] day extension.'
- AND the comment will display the timestamp of when the state change was completed
- AND the ellipsis on the right of the comment will be inactive (greyed out)

**Scenario 4: Deleted Extension system generated comment**
- GIVEN I am an IAO user
- WHEN I delete and save an extension (Public Body or OIPC) for a request
- THEN a comment will appear within the Comments tab of the request
- AND the comment will say 'Extension for _[Type - Reason]_ has been deleted.' where type = Public Body or OIPC and reason = large volume, consultation etc. (from list of extension reasons in modal)
- AND the comment will display the timestamp of when the state change was completed
- AND the ellipsis on the right of the comment will be inactive (greyed out)

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by JHarrietha-AOT** [comment]
Separated out this story based on refinement - this story can be refined with other extension stories

**Comment by arielleandrews** [comment]
@divyav-aot I have added scenarios 3-5 based on our discussion this morning and from feedback at the demo today. Can you please review and advise if you have any questions, with scenario 5 in particular. If the logic is that an approved/denied extension that is edited and approved/denied a second time, would there be two comments generated (one to say it was edited and one to say it was approved/denied)? Happy to chat this out on a call as well

**Comment by divyav-aot** [comment]
@arielleandrews @JHarrietha-AOT - I have a query on below scenario.
I edited an OIPC approved extension to Public Body extension. What is the expected comment?
Currently it shows "Extension for [OIPC - Reason] has been edited." Could you please confirm.

**Comment by JHarrietha-AOT** [comment]
@divyav-aot Thanks for the great questions.

Comment for change of extension type: "Extensions for (OPIC/PB Reason) has been edited to (OPIC/PB Reason)"
This provides better information for the user.

Messaging for OPIC Extensions Pending. "PENDING Extension for (OPIC Reason) added"

cc: @arielleandrews for final approval and rewording

**Comment by arielleandrews** [comment]
@divyav-aot Scenario 5 has been removed (comment for editing). Thank you for raising!

**Comment by divyav-aot** [comment]
Thanks @arielleandrews . As we discussed, the reason change and pending is not capturing as a comment. This includes approved extension reason change. Please correct me if I am wrong.

**Comment by arielleandrews** [comment]
Correct @divyav-aot!

**Comment by JHarrietha-AOT** [comment]
@divyav-aot Was just looking at this and noticed that some extension comments have the three dots appearing but are disabled and some don't have the 3 dots at all. For consistency I think they should either not have them unless editable or all have them with those that can't be disabled. Open to whichever is easier for you, just want them to be consistent.  ![image.png](https://images.zenhubusercontent.com/6081a3ba692e484184671a59/b76afc39-9797-46c8-a7a1-711019ff9b0d)

**Comment by JHarrietha-AOT** [comment]
@divyav-aot Let me know how you would like to tackle the above comment, happy to make a separate bug if thats is easier.

**Comment by divyav-aot** [comment]
@JHarrietha-AOT  - could you please create a separate bug. I think it is an existing bug. There were no changes to FE as part of this US.

**Comment by JHarrietha-AOT** [comment]
@divyav-aot Sounds good, I'll check to see if a bug has already been made for this, if not I will make one. Will move this to Ready for QA now.

**Comment by arielleandrews** [comment]
QA completed, all scenarios passed. Nice work @divyav-aot
